### PR TITLE
add repository address normalisation

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

File: lib/Basics/StringUtils.cpp [hunk 1]
```diff
@@ -698,16 +698,15 @@ void tolowerInPlace(std::string* str) {
...
-    str[i] = static_cast<char>(::tolower(str[i]));
-  }
+  std::transform(
+    str.begin(), str.end(), str.begin(), [](unsigned char c){ return std::tolower(c); });
 
   return std::move(str);
 }
 
 std::string tolower(std::string const& str) {
+
```
Comment (maierlars) [inline lib/Basics/StringUtils.cpp]: ⠀

File: tests/HotBackup/HotBackupCoordinatorTest.cpp [hunk 2]
```diff
@@ -339,4 +345,126 @@ TEST_F(HotBackupOnCoordinators, test_irrelevance_of_string_size_for_dbserver_id)
...
+  prefix = "/";
+  result = RClone::normalizeRepositoryString(config, prefix, repo);
+  ASSERT_EQ(result.errorNumber(), TRI_ERROR_NO_ERROR);
+  ASSERT_EQ(repo, "local:/");
+
+  repo = "local:/a/b/c";
+  prefix = "/b";
+  result = RClone::normalizeRepositoryString(config, prefix, repo);
+  ASSERT_EQ(result.errorNumber(), TRI_ERROR_REMOTE_REPOSITORY_CONFIG_BAD);
+
```
Comment (maierlars) [inline tests/HotBackup/HotBackupCoordinatorTest.cpp]: Missing `ASSERT_EQ`.

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6542/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6544/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6549/

**Pull request merged**